test/test_xla_sharding.py

```diff
@@ -147,6 +147,29 @@ def test_metrics_recorded(self):
     self.assertIn("VirtualDeviceUsage", met.counter_names())
     self.assertNotEqual(met.counter_value("VirtualDeviceUsage"), 0)
 
+  def test_outbound_data_metrics(self):
+    baseline_outbound_data = met.counter_value("OutboundData")
+    os.environ["XLA_USE_SPMD"] = "1"
+    partition_spec = (0, 1)
+    xt1 = torch.tensor([[1, 2, 3, 4, 5, 6, 7, 8]],
+                       dtype=torch.float,
+                       device=xm.xla_device())
```

Comment:
maybe add a check before `mark_sharding` and verify `OutboundData` is 0? I think it is easier to use `met.clear_counters` or `met.clear_all()` in the beginning of the test if you want to check counter value, unless you have a specified reason to keep the metrics across tests.

Reply:
Maybe add `met.clear_all()` in `SetUp` and call it a day -- since `met.clear_all()` is assumed to be working/testsed in this scope.

Reply:
I see `met.clear_counters()` but not `met.clear_all()`, and I don't think clearing counters will affect this metric. We can look at the baseline data transferred before anything gets done in the test, and then take the difference with snapshots after operations with/without virtual device, so that we can look at the data transfer delta after each type of operation. It should be equivalent.

Reply:
you can get `clear_all` by rebasing.